src/main/java/org/broadinstitute/consent/http/service/MetricsService.java

```diff
@@ -160,4 +194,54 @@ public String getHeaderRow(Type type) {
     }
     return election;
   }
+
+  public DatasetMetrics generateDatasetMetrics(Integer datasetId) {
+
+    DatasetMetrics metrics = new DatasetMetrics();
+
+    //get datasetDTO with properties and data use restrictions
+    Set<DatasetDTO> datasets = dataSetDAO.findDatasetDTOWithPropertiesByDatasetId(datasetId);
+    if (datasets == null || datasets.isEmpty()) {
+       throw new NotFoundException("Dataset with specified ID does not exist.");
+    }
+
+    //find dars with the given datasetId in their list of datasetIds, datasetId is a String so it can be converted to jsonb in query
+    //convert all dars into smaller objects that only contain the information needed
+    List<DataAccessRequest> dars = dataAccessRequestDAO.findAllDataAccessRequestsForDatasetMetrics(Integer.toString(datasetId));
+    List<DarMetricsSummary> darInfo = dars.stream().map(dar -> 
+      new DarMetricsSummary(dar.getUpdateDate(), dar.data.getProjectTitle(), dar.data.getDarCode(), dar.data.getNonTechRus(), findPI(dar.userId), dar.getReferenceId()))
+      .collect(Collectors.toList());
+
+    //if there are associated dars, find associated access elections so we know how many and which dars are approved/denied
+    List<String> referenceIds = dars.stream().map(dar -> (dar.referenceId)).collect(Collectors.toList());
+    if (!referenceIds.isEmpty()) {
+      List<Election> elections = electionDAO.findLastElectionsByReferenceIdsAndType(referenceIds, "DataAccess");
+      metrics.setElections(elections);
+    } else {
+      metrics.setElections(Collections.emptyList());
+    }
+    metrics.setDataset(datasets.iterator().next());
+    metrics.setDars(darInfo);
+    return metrics;
+
+  }
+
+  public String findPI(Integer userId) {
+    List<UserProperty> props = userPropertyDAO.findResearcherPropertiesByUser(userId);
+
+    Optional<UserProperty> isResearcher = props.stream().filter(prop -> prop.getPropertyKey().equals("isThePI") && prop.getPropertyValue().toLowerCase().equals("true")).findFirst();
+    if (isResearcher.isPresent()) {
+      Optional<UserProperty> userName = props.stream().filter(prop -> prop.getPropertyKey().equals("profileName")).findFirst();
```

Comment:
We have a display name right on the user, so you don't need to go through the properties to find that value.

Reply:
I don't think I have the user object. I looked for a DAO call that got a user and its properties by its userId but I don't see any like that in either the userDAO or the userPropertyDAO, so instead, I only get the user properties using the userpropertyDAO call

Reply:
No, you don't have the user object, but you do have the user id and can query on it. We could add a dao call to get the user with properties as well. We really should be looking directly at the user since the `profileName` is a brittle value. We need a story to deprecate some of these. Looking at `UserFields.PROFILE_NAME`, we never actually use that to **_set_** a property value, we only use it to **_get_** a property value.

Reply:
okay makes sense! I will add a new DAO call

Reply:
We do use the `profileName` value in the UI code though - we should clean this up these usages to all be using the same field.

Reply:
Okay, I will file a ticket for that!

Reply:
For this PR, let me rephrase my request that we make the constructor easier to use. With respect to the user profile name, we can address that separately in a different PR.